### PR TITLE
private repo support

### PR DESCRIPTION
**Pull request by FrimIdan**

### CODE REVIEW

File: docker/token/token.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		if err != nil {
+			log.Debugf("failed to get credentials. image=%v: %v", named.Name(), err)
+			continue
+		}
+
+		log.Debugf("Credentials found. image name=%v, extractor=%v", named.Name(), extractor.Name())
+		return username, password, nil
+	}
+
+	return "", "", nil
```
Comment (akpsgit) [inline docker/token/token.go:54]: I think we should print no match before exiting

File: docker/token/token.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+)
+
+type CredExtractor struct {
+	extractors []Extractor
+}
+
+type Extractor interface {
+	// Prints the name of the extractor
+	Name() string
+	// Returns true if extractor is matched for extracting credentials for the given image
```
Comment (akpsgit) [inline docker/token/token.go]: Returns true if extractor is supported for extracting credentials for the given image

File: docker/token/token.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	IsMatch(named reference.Named) bool
```
Comment (akpsgit) [inline docker/token/token.go]: IsSupported

File: docker/token/token.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		username, password, err = extractor.GetCredentials(ctx, named)
+		if err != nil {
+			log.Debugf("failed to get credentials. image=%v: %v", named.Name(), err)
+			continue
+		}
+
+		log.Debugf("Credentials found. image name=%v, extractor=%v", named.Name(), extractor.Name())
+		return username, password, nil
+	}
+
```
Comment (akpsgit) [inline docker/token/token.go:52]: Think we should print that no credentials found

**Review by akpsgit — COMMENTED** [review]
Looks good!

File: docker/token/token.go (hunk 3 above)
Comment (FrimIdan) [reply docker/token/token.go]: @akpsgit Why `isSupported`? We are looking for an extractor that matches the image. ex. gcr extractor matches only gcr.io images..

File: docker/token/token.go (hunk 4 above)
Comment (FrimIdan) [reply docker/token/token.go:52]: Yes, I got this in the previous comment (:

**Pull request merged**